projects/portal/src/app/pages/faucet/faucet.usecase.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Config, ConfigService } from 'projects/portal/src/app/models/config.service';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { first, map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class FaucetUseCaseService {
+
+  //private config$: Observable<Config | undefined>;
+
+  constructor(
+    private readonly route: ActivatedRoute,
```

Comment:
I would like to DI routing related dependencies such as ActivatedRoute with Component. Sorry I didn't say that.

Reason
- Make it easier to test UsecaseService.
- Component's responsibility is to connect Routing, Logic,  and View

Reply:
Ok,
I have puestion. `creditAmount$` needs  `ActivatedRoute` info.
How do I give the `ActivatedRoute` value from the component to this `faucet.usecase.service.ts`?

Reply:
Good question.
Change it from a getter to a method and give it the required value as an argument.
I will comment the specific proposed modification in the appropriate section.

Reply:
Giving a value as argument, I couldn't notice that, but it was simple solution.
Thank you. It works.